pkg/certs/certs.go

```diff
@@ -78,20 +73,20 @@ func New(certFile, keyFile string, loadCert LoadX509KeyPairFunc) (*Certs, error)
 	return c, nil
 }
 
-func checkSymlink(file string) (bool, error) {
+func checkSymlink(file string) bool {
 	st, err := os.Lstat(file)
 	if err != nil {
-		return false, err
+		return false
 	}
-	return st.Mode()&os.ModeSymlink == os.ModeSymlink, nil
+	return st.Mode()&os.ModeSymlink == os.ModeSymlink
 }
 
 // watchSymlinks reloads symlinked files since fsnotify cannot watch
 // on symbolic links.
 func (c *Certs) watchSymlinks() (err error) {
 	c.Lock()
-	c.cert, err = c.loadCert(c.certFile, c.keyFile)
-	c.Unlock()
+	defer c.Unlock()
```

Comment:
This defer will deadlock locally, you should keep the previous code.

Reply:
I'm able to see why this should deadlock?! Note that the `for {...}` loop is executed in a separate go routine. Deadlock should only happen if either all go routines are trying to get the lock or the same go routine tries to acquire the lock twice... Further the `Lock(), defer Unlock()` is the recommended approach by idiomatic Go (whenever possible).

Reply:
The coding style previously was to ensure that we are only locking situations which are need to be locked. Doing defer style is idiomatic I agree but it is a bigger region lock which is not needed here. 

We should avoid cleanups in a PR as much as possible, where it's not needed. This is one such place.